src/project.cc

```diff
@@ -603,12 +603,25 @@ std::vector<Project::Entry> LoadCompilationEntriesFromDirectory(
 
   EnsureEndsInSlash(comp_db_dir);
 
-  LOG_S(INFO) << "Trying to load " << comp_db_dir << "compile_commands.json";
-  // Do not call clang_CompilationDatabase_fromDirectory if
-  // compile_commands.json does not exist; it will report an error on stderr.
-  if (FileExists(comp_db_dir + "compile_commands.json")) {
-    cx_db = clang_CompilationDatabase_fromDirectory(comp_db_dir.c_str(),
-                                                    &cx_db_load_error);
+  // This should probably be a bit more configurable...
+  std::vector<std::string> subdirs;
+  subdirs.emplace_back("");
+  if (opt_compilation_db_dir.empty()) {
+    subdirs.emplace_back("build/");
+  }
+
+  for (auto& d : subdirs) {
+    std::string temp_db_dir = comp_db_dir + d;
+    LOG_S(INFO) << "Trying to load " << temp_db_dir << "compile_commands.json";
+    // Do not call clang_CompilationDatabase_fromDirectory if
+    // compile_commands.json does not exist; it will report an error on stderr.
+    if (FileExists(temp_db_dir + "compile_commands.json")) {
```

Comment:
Since this is now a for loop we can make this an early return, ie,

```
if (!FileExists(...))
  continue;
```

Reply:
Hmm. My mental model was that `compilationDatabaseDirectory` (if set) would always just be used, and there'd be a list of other directories (relative to the project root), and of those the first one containing a `compile_commands.json` would win. (Admittedly that's not exactly what I coded, since the loop will continue if such a file exists but doesn't produce a database for some reason.)
We could invert the test as you suggest but I'm not sure that that really makes the loop clearer, does it? Presuming I dropped the test on the returned database (which probably makes sense) we'd end up with something like

    for (auto& ...) {
       std::string temp_db_dir = ...;
       if (!FileExists(...))
         continue;
       cx_db = clang_CompilationDatabase_fromDirectory(...);
       break;
    }

It's a matter of style, I guess, but that looks slightly weird. (Not horrible, just a bit weird.) I think it'd be more natural to have it more like the way I have it (though probably without the test on cx_db)), so

      if (FileExists(...)) {
          cx_db = ...;
          break;
      }

Reply:
Indeed, this would subtly changing the `compilationDatabaseDirectory` semantics. I believe we're "upgrading" the power of the option, and existing users should experience no changes. I think the choice between checking local dir or `compilationDatabaseDirectory` first is sort of arbitrary, so I'll leave it up to you. (Note that we do not recurse on local dir).

The reason to invert the loop is following a general coding style: handle errors first, then continue. This avoids high levels of indentation. ie,
```
void foo() {
  if (someError1)
    return;
  if (someError2)
    return;
  // logic that runs when there are no errors
}
```

Of course style is never perfect, either one is fine for this case.

Reply:
On reflection, I think maybe your first idea was right: just make `build` the default `compilationDatabaseDirectory` and try both that and the project root.
So something like `change-comp_db_dir`, 6a1339db878a25af8fba91ca9ee5d1de30237c18 (two commits)? (More lines changed I think, but likely easier behaviour for users to follow.)

Reply:
> On reflection, I think maybe your first idea was right: just make build the default compilationDatabaseDirectory and try both that and the project root.

Yes, this sounds good to me. Let me know when I should take another look at the PR.